server/service/src/processors/transfer/requisition/test.rs

```diff
@@ -90,41 +90,54 @@ async fn requisition_transfer() {
     let test_handle = exec_concurrent(
         test_input,
         number_of_instances,
-        |_, test_input| async move {
+        |thread_num, test_input| async move {
             let (service_provider, request_store, response_store, item1, item2) = test_input;
 
             let ctx = service_provider.basic_context().unwrap();
+            let mut tester = RequisitionTransferTester::new(
+                &request_store,
+                &response_store,
+                &item1,
+                &item2,
+                thread_num,
+            );
 
-            let mut tester =
-                RequisitionTransferTester::new(&request_store, &response_store, &item1, &item2);
-
+            log::debug!("{}: insert", thread_num);
             tester.insert_request_requisition(&ctx.connection);
             // manually trigger because inserting the requisition doesn't trigger the processor
             ctx.processors_trigger
                 .requisition_transfer
                 .try_send(())
                 .unwrap();
+            log::debug!("{}: await_events_processed", thread_num);
             ctx.processors_trigger.await_events_processed().await;
+            log::debug!("{}: check_response_requisition_not_created", thread_num);
             tester.check_response_requisition_not_created(&ctx.connection);
+            log::debug!("{}: update_request_requisition_to_sent", thread_num);
             tester.update_request_requisition_to_sent(&service_provider);
             ctx.processors_trigger.await_events_processed().await;
-            tester.check_response_requisition_created(&ctx.connection);
+            log::debug!("{}: check_response_requisition_created", thread_num);
+            tester.check_response_requisition_created(&ctx.connection, thread_num);
             ctx.processors_trigger.await_events_processed().await;
+            log::debug!("{}: check_request_requisition_was_linked", thread_num);
             tester.check_request_requisition_was_linked(&ctx.connection);
-            tester.update_response_requisition_to_approved(&service_provider);
+            log::debug!("{}: update_response_requisition_to_approved", thread_num);
+            tester.update_response_requisition_to_approved(&service_provider, thread_num);
             // Response requisition approval is usually done by mSupply
             // Processor would be triggered after sync
             // We've approved manually for testing, so need to manually trigger the processor as well
             ctx.processors_trigger
                 .requisition_transfer
                 .try_send(())
                 .unwrap();
+            log::debug!("{}: await_events_processed", thread_num);
             ctx.processors_trigger.await_events_processed().await;
-            // Sometimes processor isn't finished the first time, so we await twice?
```

Comment:
Took this out, as it seems like it was an attempt to fix something, but might be masking an issue?